sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2044,6 +2044,16 @@ class Dataset[T] private[sql](
     Limit(Literal(n), logicalPlan)
   }
 
+  /**
+   * Returns a new Dataset by skipping the first `m` rows and then taking the followed `n` rows.
+   *
+   * @group typedrel
+   * @since 3.4.0
+   */
+  def limitAndOffset(n: Int, m: Int): Dataset[T] = withTypedPlan {
```

Comment:
This does not match the SQL API. Shall we just have `def offset(...)`?

Reply:
https://github.com/apache/spark/pull/36319 move the check of Offset from analysis to finsh analysis, so we can add `def offset(...)`.